### ISSUE TITLE
xml visibleCondition="false" not working as expected

### ISSUE THREAD

**Issue by etidimjav** [sulu/sulu#5539]
| Q | A
| --- | ---
| Sulu Version | 2.1.2

# Actual Behavior

When we put for a property the visibleCondition to false, the input is still display. False seems to not be treat as boolean.

# Expected Behavior

Keywords true and false should be treat as boolean.

**Comment by nnatter** [comment]
I just had a quick look at this. `<property ... visibleCondition="false">` is indeed not working at the moment. The problem here is that `visibleCondition="false"` is parsed as boolean value from the XML. After parsing, it is then implicitly casted to `''` because of the string typehint of the `ItemMetadata::setVisibleCondition` method in the following line:

https://github.com/sulu/sulu/blob/229d861d93af610058e7a7374a062ee779f4b67d/src/Sulu/Component/Content/Metadata/Parser/PropertiesXmlParser.php#L416

Something like this should be enough to fix the issue:

```diff
--- a/src/Sulu/Component/Content/Metadata/Parser/PropertiesXmlParser.php
+++ b/src/Sulu/Component/Content/Metadata/Parser/PropertiesXmlParser.php
@@ -353,11 +353,11 @@ class PropertiesXmlParser
         }

         if (isset($data['disabledCondition'])) {
-            $section->setDisabledCondition($data['disabledCondition']);
+            $section->setDisabledCondition($this->normalizeConditionData($data['disabledCondition']));
         }

         if (isset($data['visibleCondition'])) {
-            $section->setVisibleCondition($data['visibleCondition']);
+            $section->setVisibleCondition($this->normalizeConditionData($data['visibleCondition']));
         }

         foreach ($data['properties'] as $name => $property) {
@@ -373,11 +373,11 @@ class PropertiesXmlParser
         $blockProperty->setName($propertyName);

         if (isset($data['disabledCondition'])) {
-            $blockProperty->setDisabledCondition($data['disabledCondition']);
+            $blockProperty->setDisabledCondition($this->normalizeConditionData($data['disabledCondition']));
         }

         if (isset($data['visibleCondition'])) {
-            $blockProperty->setVisibleCondition($data['visibleCondition']);
+            $blockProperty->setVisibleCondition($this->normalizeConditionData($data['visibleCondition']));
         }

         if (isset($data['meta']['title'])) {
@@ -409,12 +409,8 @@ class PropertiesXmlParser
         $property->setTags($data['tags']);
         $property->setMinOccurs(null !== $data['minOccurs'] ? \intval($data['minOccurs']) : null);
         $property->setMaxOccurs(null !== $data['maxOccurs'] ? \intval($data['maxOccurs']) : null);
-        $property->setDisabledCondition(
-            \array_key_exists('disabledCondition', $data) ? $data['disabledCondition'] : null
-        );
-        $property->setVisibleCondition(
-            \array_key_exists('visibleCondition', $data) ? $data['visibleCondition'] : null
-        );
+        $property->setDisabledCondition($this->normalizeConditionData($data['disabledCondition'] ?? null));
+        $property->setVisibleCondition($this->normalizeConditionData($data['visibleCondition'] ?? null));
         $property->setParameters($data['params']);
         $property->setOnInvalid(\array_key_exists('onInvalid', $data) ? $data['onInvalid'] : null);
         $this->mapMeta($property, $data['meta']);
@@ -458,6 +454,15 @@ class PropertiesXmlParser
         return $data;
     }

+    private function normalizeConditionData($data): ?string
+    {
+        if (is_bool($data)) {
+            return $data ? 'true' : 'false';
+        }
+
+        return $data;
+    }
+
     private function normalizeItem($data): array
     {
         $data = \array_merge_recursive(
```

**Comment by danrot** [comment]
Fixed in #5585
